mergify_engine/context.py

```diff
@@ -100,6 +100,24 @@ async def get_user_tokens(self) -> user_tokens.UserTokens:
             )
         return self._user_tokens
 
+    USER_ID_MAPPING_CACHE_KEY: str = "user-id-mapping"
+
+    async def get_user_id(
+        self, login: github_types.GitHubLogin
+    ) -> github_types.GitHubAccountIdType:
+        user_id = typing.cast(
+            typing.Optional[github_types.GitHubAccountIdType],
+            await self.redis.hget(self.USER_ID_MAPPING_CACHE_KEY, login),
+        )
+        if user_id is not None:
+            return user_id
+
+        user = typing.cast(
+            github_types.GitHubAccount, await self.client.item(f"/users/{login}")
+        )
+        await self.redis.hset(self.USER_ID_MAPPING_CACHE_KEY, login, user["id"])
```

Comment:
What about caching the whole `user` dataset?
There's a chance we need another field at some point 🤷

Reply:
hum, why not